cirq/qis/states.py

```diff
@@ -93,66 +115,81 @@ def density_matrix_from_state_vector(
         $$
 
     Args:
-        state: A sequence representing a wave function in which
+        state_vector: A sequence representing a state vector in which
             the ordering mapping to qubits follows the standard Kronecker
-            convention of numpy.kron.
+            convention of numpy.kron (big-endian).
         indices: list containing indices for qubits that you would like
             to include in the density matrix (i.e.) qubits that WON'T
             be traced out. follows the standard Kronecker convention of
             numpy.kron.
+        qid_shape: specifies the dimensions of the qudits for the input
+            `state_vector`.  If not specified, qubits are assumed and the
+            `state_vector` must have a dimension a power of two.
 
     Returns:
         A numpy array representing the density matrix.
 
     Raises:
-        ValueError: if the size of state is not a power of 2.
-        ValueError: if the size of the state represents more than 25 qubits.
+        ValueError: if the size of `state_vector` is not a power of 2 and the
+            shape is not given or if the shape is given and `state_vector`
+            has a size that contradicts this shape.
         IndexError: if the indices are out of range for the number of qubits
-            corresponding to the state.
+            corresponding to `state_vector`.
     """
-    qid_shape = validate_qid_shape(state, qid_shape)
-    n_qubits = len(qid_shape)
+    shape = validate_qid_shape(state_vector, qid_shape)
+    n_qubits = len(shape)
 
     if indices is None:
-        return np.outer(state, np.conj(state))
+        return np.outer(state_vector, np.conj(state_vector))
 
     indices = list(indices)
     validate_indices(n_qubits, indices)
 
-    state = np.asarray(state).reshape(qid_shape)
+    state_vector = np.asarray(state_vector).reshape(shape)
 
     sum_inds = np.array(range(n_qubits))
     sum_inds[indices] += n_qubits
 
-    rho = np.einsum(state, list(range(n_qubits)), np.conj(state),
+    rho = np.einsum(state_vector, list(range(n_qubits)), np.conj(state_vector),
                     sum_inds.tolist(), indices + sum_inds[indices].tolist())
-    new_shape = np.prod([qid_shape[i] for i in indices], dtype=int)
```

Comment:
Unrelated rename?

Reply:
No, there is a bad no no aboe: qid_shape = validate_qid_shate(state, qid_state) could modify qid_shape.  Caught by linter, not sure why it wasn't picked up earlier.